### PR TITLE
[Fix JENKINS-44663] Make Basic authorization header to be case in-sensitive

### PR DESCRIPTION
**Pull request by liketic**
See [JENKINS-44663](https://issues.jenkins-ci.org/browse/JENKINS-44663).

### Proposed changelog entries

* BUG: Make BASIC authorization header to be case in-sensitive
* ...

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

**Review by daniel-beck — APPROVED** [review]

File: test/src/test/java/jenkins/security/BasicHeaderProcessorTest.java [hunk 1]
```diff
@@ -81,13 +81,39 @@ private void makeRequestAndFail(String userAndPass) throws IOException, SAXExcep
     }
 
     private void makeRequestWithAuthAndVerify(String userAndPass, String username) throws IOException, SAXException {
+        String authCode = "Basic "+Scrambler.scramble(userAndPass);
+        doMakeRequestWithAuthAndVerify(authCode, username);
+    }
+
+    private void doMakeRequestWithAuthAndVerify(String authCode, String expected) throws IOException, SAXException {
```
Comment (oleg-nenashev) [inline test/src/test/java/jenkins/security/BasicHeaderProcessorTest.java]: doMake... sounds like a Stapler endpoint prefix, but it's does not matter much for a private method

**Review by oleg-nenashev — APPROVED** [review]
👍

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Hmm, test failure is definitely related...

**Comment by liketic** [comment]
@oleg-nenashev Thank you. I didn't figure out how to run a single test yet, any help is appreciated. :) .

**Comment by oleg-nenashev** [comment]
@liketic Thanks for the fix. In the case of Java/Groovy tests in the repo, see this page: http://maven.apache.org/surefire/maven-surefire-plugin/examples/single-test.html

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**